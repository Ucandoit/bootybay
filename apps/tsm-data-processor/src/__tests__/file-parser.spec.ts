import { join } from 'path';
import FileParser from '../file-parser';

describe('FileParser', () => {
  describe('parse', () => {
    const fileParser = new FileParser();
    test('it should throw error if the file is not fould', () => {
      expect(() => {
        fileParser.parse(join(__dirname, 'samples/bad'), 'bcc-eu-0123456789');
      }).toThrowError();
    });

    test('it should throw error if file name has incorrect syntax', () => {
      expect(() => {
        fileParser.parse(join(__dirname, 'samples/bad'), 'bcc-eu.txt');
      }).toThrowError('File name syntax error');
    });

    test('it should throw error if timestamp is not in file name', () => {
      expect(() => {
        fileParser.parse(join(__dirname, 'samples/bad'), 'bcc-eu-test.txt');
      }).toThrowError('Parse download time error');
    });

    test('it should throw error if data is not present in the file', () => {
      expect(() => {
        fileParser.parse(join(__dirname, 'samples/bad/'), 'bcc-eu-1639008718.txt');
      }).toThrowError('Data not found');
    });

    test('it should return a list if file is read correctly', () => {
      const regionalAuctions = fileParser.parse(join(__dirname, 'samples/good'), 'bcc-eu-1639008717.txt');
      expect(regionalAuctions).toMatchSnapshot();
      const realmAuctions = fileParser.parse(join(__dirname, 'samples/good'), 'bcc-eu-mograine-horde-1639038664.txt');
      expect(realmAuctions).toMatchSnapshot();
    });

    test('it should throw error if data format is not correct in the file', () => {
      expect(() => {
        fileParser.parse(join(__dirname, 'samples/bad/'), 'bcc-eu-1639008719.txt');
      }).toThrowError('Data format incorrect');
    });
  });
});
