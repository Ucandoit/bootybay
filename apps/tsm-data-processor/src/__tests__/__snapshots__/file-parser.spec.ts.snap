// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileParser parse it should return a list if file is read correctly 1`] = `
Array [
  Object {
    "itemString": "25005",
    "realm": "bcc-eu",
    "regionHistorical": 186323,
    "regionMarketValue": 210814,
    "regionSale": 117248,
    "regionSalePercent": 8,
    "regionSoldPerDay": 8,
    "timestamp": 1639008717,
  },
  Object {
    "itemString": "4790",
    "realm": "bcc-eu",
    "regionHistorical": 28698,
    "regionMarketValue": 30974,
    "regionSale": 6266,
    "regionSalePercent": 1,
    "regionSoldPerDay": 0,
    "timestamp": 1639008717,
  },
  Object {
    "itemString": "23626",
    "realm": "bcc-eu",
    "regionHistorical": 5776824,
    "regionMarketValue": 5450345,
    "regionSale": 3149950,
    "regionSalePercent": 5,
    "regionSoldPerDay": 0,
    "timestamp": 1639008717,
  },
]
`;

exports[`FileParser parse it should return a list if file is read correctly 2`] = `
Array [
  Object {
    "historical": 50000,
    "itemString": "38",
    "marketValue": 50000,
    "minBuyout": 50000,
    "numAuctions": 2,
    "realm": "bcc-eu-mograine-horde",
    "timestamp": 1639038664,
  },
  Object {
    "historical": 200000,
    "itemString": "39",
    "marketValue": 200000,
    "minBuyout": 0,
    "numAuctions": 0,
    "realm": "bcc-eu-mograine-horde",
    "timestamp": 1639038664,
  },
  Object {
    "historical": 200000,
    "itemString": "43",
    "marketValue": 200000,
    "minBuyout": 0,
    "numAuctions": 0,
    "realm": "bcc-eu-mograine-horde",
    "timestamp": 1639038664,
  },
]
`;
